flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/taskmanager/TaskManagerDetailsHandlerTest.java

```diff
@@ -0,0 +1,216 @@
+
+	private TestingResourceManagerGateway resourceManagerGateway;
+	private TaskManagerDetailsHandler testInstance;
+
+	private HandlerRequest<EmptyRequestBody, TaskManagerMessageParameters> handlerRequest;
+
+	public void initializeMetricStore(MetricStore metricStore) {
+		QueryScopeInfo.TaskManagerQueryScopeInfo tmScope = new QueryScopeInfo.TaskManagerQueryScopeInfo(taskManagerId.toString(), "Status");
+
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Heap.Used", 1));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Heap.Committed", 2));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Heap.Max", 3));
+
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.NonHeap.Used", 4));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.NonHeap.Committed", 5));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.NonHeap.Max", 6));
+
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Direct.Count", 7));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Direct.MemoryUsed", 8));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Direct.TotalCapacity", 9));
+
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Mapped.Count", 10));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Mapped.MemoryUsed", 11));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "JVM.Memory.Mapped.TotalCapacity", 12));
+
+		metricStore.add(new MetricDump.CounterDump(tmScope, "Network.AvailableMemorySegments", 13));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "Network.TotalMemorySegments", 14));
+
+		metricStore.add(new MetricDump.CounterDump(tmScope, "Shuffle.Netty.AvailableMemorySegments", 15));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "Shuffle.Netty.UsedMemorySegments", 16));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "Shuffle.Netty.TotalMemorySegments", 17));
+
+		metricStore.add(new MetricDump.CounterDump(tmScope, "Shuffle.Netty.AvailableMemory", 18));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "Shuffle.Netty.UsedMemory", 19));
+		metricStore.add(new MetricDump.CounterDump(tmScope, "Shuffle.Netty.TotalMemory", 20));
+	}
+
+	private HandlerRequest<EmptyRequestBody, TaskManagerMessageParameters> createRequest() throws HandlerRequestException {
+		Map<String, String> pathParamaters = new HashMap<>();
+		pathParamaters.put(TaskManagerIdPathParameter.KEY, taskManagerId.toString());
+
+		return new HandlerRequest<>(
+			EmptyRequestBody.getInstance(),
+			new TaskManagerFileMessageParameters(),
+			pathParamaters,
+			Collections.emptyMap()
+		);
+	}
+
+	@Before
+	public void setup() throws HandlerRequestException {
+		resourceManagerGateway = new TestingResourceManagerGateway();
+
+		MetricFetcher metricFetcher = new TestingMetricFetcher();
+		initializeMetricStore(metricFetcher.getMetricStore());
```

Comment:
the called method `initializeMetricStore` declaration proceeds the calling method